File: lib/utils/commons.dart
import 'package:floaty_head/floaty_head.dart';

class Commons {
  /// Replace the [windowGravity] setted in dart-client code.
  static String getWindowGravity(FloatyHeadGravity? gravity) {
    if (gravity == null) gravity = FloatyHeadGravity.top;
    switch (gravity) {
      case FloatyHeadGravity.center:
        return "center";
      case FloatyHeadGravity.bottom:
        return "bottom";
      case FloatyHeadGravity.top:
      default:
        return "top";
    }
  }

  /// Replace the [contentGravity] setted in dart-client code.
  static String getContentGravity(ContentGravity? gravity) {
    if (gravity == null) gravity = ContentGravity.left;
    switch (gravity) {
      case ContentGravity.center:
        return "center";
      case ContentGravity.right:
        return "right";
      case ContentGravity.left:
      default:
        return "left";
    }
  }

  /// Replace the [position] setted in dart-client code.
  static String getPosition(ButtonPosition? buttonPosition) {
    if (buttonPosition == null) buttonPosition = ButtonPosition.center;
    switch (buttonPosition) {
      case ButtonPosition.leading:
        return "leading";
      case ButtonPosition.trailing:
        return "trailing";
      case ButtonPosition.center:
      default:
        return "center";
    }
  }

  /// Replace the [fontWeight] setted in dart-client code.
  static String getFontWeight(FontWeight? fontWeight) {
    if (fontWeight == null) fontWeight = FontWeight.normal;
    switch (fontWeight) {
      case FontWeight.bold:
        return "bold";
      case FontWeight.italic:
        return "italic";
      case FontWeight.bold_italic:
        return "bold_italic";
      case FontWeight.normal:
      default:
        return "normal";
    }
  }
}

File: android/src/main/kotlin/ni/devotion/floaty_head/utils/Commons.kt
package ni.devotion.floaty_head.utils

import android.content.Context
import android.graphics.Typeface
import android.util.TypedValue
import android.view.Gravity
import android.widget.LinearLayout
import androidx.annotation.Nullable
import ni.devotion.floaty_head.models.Margin
import ni.devotion.floaty_head.utils.Constants.KEY_MARGIN


object Commons {
    fun getMapFromObject(map: Map<String, Any>, key: String?): Map<String, Any>? {
        return map[key] as Map<String, Any>?
    }

    fun getMapListFromObject(map: Map<String, Any>, key: String?): List<Map<String, Any>>? {
        return map[key] as List<Map<String, Any>>?
    }

    fun getSpFromPixels(context: Context, px: Float): Float {
        val scaledDensity = context.resources.displayMetrics.scaledDensity
        return px / scaledDensity
    }

    fun getPixelsFromDp(context: Context, dp: Int): Int {
        return if (dp == -1) -1 else TypedValue.applyDimension(
                TypedValue.COMPLEX_UNIT_DIP, dp.toFloat(), context.resources.displayMetrics).toInt()
    }

    fun getPixelsFromDp(context: Context, dp: Float): Float {
        return if (dp == -1f) (-1).toFloat() else TypedValue.applyDimension(
                TypedValue.COMPLEX_UNIT_DIP, dp, context.resources.displayMetrics)
    }

    fun getGravity(@Nullable gravityStr: String?, defVal: Int): Int {
        var gravity = defVal
        if (gravityStr != null) {
            when (gravityStr) {
                "top" -> gravity = Gravity.TOP
                "center" -> gravity = Gravity.CENTER
                "bottom" -> gravity = Gravity.BOTTOM
                "leading" -> gravity = Gravity.START
                "trailing" -> gravity = Gravity.END
            }
        }
        return gravity
    }

    fun getFontWeight(@Nullable fontWeightStr: String?, defVal: Int): Int {
        var fontWeight = defVal
        if (fontWeightStr != null) {
            fontWeight = when (fontWeightStr) {
                "normal" -> Typeface.NORMAL
                "bold" -> Typeface.BOLD
                "italic" -> Typeface.ITALIC
                "bold_italic" -> Typeface.BOLD_ITALIC
                else -> Typeface.NORMAL
            }
        }
        return fontWeight
    }

    fun setMargin(context: Context?, params: LinearLayout.LayoutParams, map: Map<String, Any?>) {
        val margin: Margin = UiBuilder.getMargin(context, map[KEY_MARGIN])
        params.setMargins(margin.left, margin.top, margin.right, margin.bottom)
    }
}
